promql/promqltest/test.go

```diff
@@ -450,15 +520,35 @@ func (*test) parseEval(lines []string, i int) (int, *evalCmd, error) {
 			return i, nil, err
 		}
 
-		// Currently, we are not expecting any matrices.
-		if len(vals) > 1 && isInstant {
-			return i, nil, formatErr("expecting multiple values in instant evaluation not allowed")
+		// Only allow a range vector for an instant query where we have defined the expected range vector timestamps.
+		if len(vals) > 1 && isInstant && !expectRangeVector {
+			return i, nil, formatErr("expecting multiple values in instant evaluation not allowed. consider using 'expect range vector' directive to enable a range vector result for an instant query")
 		}
 		cmd.expectMetric(j, metric, vals...)
 	}
 	return i, cmd, nil
 }
 
+// parseAsStringLiteral returns the expected string from an expect string expression.
+// It is valid for the line to match the expect string prefix exactly, and an empty string is returned.
+func parseAsStringLiteral(line string) (string, error) {
+	if line == expectStringPrefix {
+		return "", nil
+	}
+
+	parts := patExpectString.FindStringSubmatch(line)
+	if len(parts) != 2 {
+		return "", errors.New("expected string literal not valid")
+	}
+
+	str, err := strconv.Unquote(parts[1])
+	if err != nil {
+		return "", errors.New("expected string literal must be within quotes")
```

Comment:
Nit: this is not the only way in which `Unquote` can fail; for instance an invalid quote like `\c`.
(but it doesn't tell you why it failed)